[Ping Zhi] iP

TanPingZhi opened the pull request:
DukePro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is FREE!

Features:

- [X] Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```
Future improvements:
- [x] Add more exception handling
- [x] Add more junit tests

Commit:
Add Gradle support

TanPingZhi committed:
set up duke

TanPingZhi committed:
added task list

TanPingZhi committed:
created cmdhandler

TanPingZhi committed:
created child classes of task

TanPingZhi committed:
added testing

TanPingZhi committed:
added exceptions to handle todo and invalid cmd

TanPingZhi committed:
added delete

TanPingZhi committed:
add data/tasks_data

data are stored in the format of
[TDE] | [ X] | desc | time

TanPingZhi committed:
Revert "add data/tasks_data"

This reverts commit ee1b46b7e6e4caad277e411ffd216f7e615f8a13.

TanPingZhi committed:
Revert "Revert "add data/tasks_data""

This reverts commit 757b9922753d502b48f469cd845829bad58a8889.

TanPingZhi committed:
Remove tasks_data.txt

Signed-off-by: TanPingZhi <josiahtpz@gmail.com>

TanPingZhi committed:
Update Event and Deadline

added localdate

TanPingZhi committed:
Add javadoc to cmdhandler

just the handleMark for testing

TanPingZhi committed:
Merge branch 'branch-Level-8'

TanPingZhi committed:
Merge branch 'branch-Level-7'

TanPingZhi committed:
Add Ui Storage Parse TaskList Command

More OOP

TanPingZhi committed:
Remove redundant imports

TanPingZhi committed:
Update gitignore

gitignore will now stop tracking tasks.txt

TanPingZhi committed:
Remove tasks.txt from cache

TanPingZhi committed:
Organize into Packages

All classes are placed into package duke

TanPingZhi committed:
Fix missing Mark and Unmark Commands bug

TanPingZhi committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

TanPingZhi committed:
Add JUnit Tests to TaskList and Parser

TaskList methods tested:
mark
unmark

Parser methods tested:
parse
commandToTask

TanPingZhi committed:
Create jar file and updated Storage

Storage creates a bug when the directory where the
tasks.txt lives do not exist

TanPingZhi committed:
Add JavaDoc to classes and methods

TanPingZhi committed:
Remove use of null in Storage

TanPingZhi committed:
Merge branch 'branch-A-CodingStandard'

TanPingZhi committed:
Update Parser with switch statements

TanPingZhi committed:
Add FindCommand

FindCommand allows the user to filter their
tasks with a given input

TanPingZhi committed:
Merge branch 'branch-Level-9'

TanPingZhi committed:
Add JavaDoc for ExitCommand

TanPingZhi committed:
Merge branch 'branch-A-JavaDoc'

TanPingZhi committed:
Update switch statement in Parser

TanPingZhi committed:
Fix whatever that was causing bugs

I do not know what happened but there was some problem with intellij

TanPingZhi committed:
Add GUI to Duke

The GUI interacts with Duke through the Duke method getResponse

TanPingZhi committed:
Update gitignore
/data/duke.txt

TanPingZhi committed:
Drop files from gitignore

TanPingZhi committed:
Add Checkstyle

TanPingZhi committed:
Merge branch 'branch-Level-10'

TanPingZhi committed:
Add Javadocs to all public classes and methods

gunbux commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,115 @@
+    static final int MARK_LENGTH = 5;
+    static final int UNMARK_LENGTH = 7;
+    static final int DEL_LENGTH = 7;
+    static final String EXIT_SYNTAX = "bye";
+    static final String LIST_SYNTAX = "list";
+    static final String DELETE_SYNTAX = "delete";
+    static final String MARK_SYNTAX = "mark";
+    static final String UNMARK_SYNTAX = "unmark";
```
Would enums be better here?

gunbux commented on src/main/java/duke/Main.java:
```diff
@@ -0,0 +1,33 @@
+ * Main entry point to run the java app
+ */
+public class Main extends Application {
+    private static final String FILE_PATH = "data/tasks.txt";
```
Noticed this is different from your file path in the Duke class, maybe can try and standardize it!

gunbux reviewed (commented):
LGTM, just minor comments (not really issues)

wxy1203 commented on src/main/java/CmdHandler.java:
```diff
@@ -0,0 +1,55 @@
+    private void handleUnMark(int i) {
+        tasks.get(i).unMark();
+        System.out.println("marked as undone: " + tasks.get(i).toStringWithStatusIcon());
+    }
```
Nice method naming and code naming style is quite clear and understandable.

wxy1203 commented on src/main/java/Duke.java:
```diff
@@ -4,45 +4,13 @@
-            }
-        }
+        CmdHandler cmdHandler = new CmdHandler();
+        cmdHandler.handle();
```
Good implementation of OOP and this exteneral class for commend handler. :smiley:

wxy1203 commented on src/main/java/CmdHandler.java:
```diff
@@ -45,20 +56,26 @@ private void handleUnMark(int i) {
+
+            } catch (DukeException de) {
+                System.out.println(de.getMessage());
+
```
The try and catch implementation of the exception handler and you consider about another exception class just to handle the command error. :+1:

wxy1203 commented on src/main/java/duke/Parser.java:
```diff
@@ -88,7 +89,10 @@ private static boolean isAddCommand(String fullCommand) {
     public static Command parse(String fullCommand) throws DukeException {
-
+        switch (getFirstWord(fullCommand)) {
+        case FIND_SYNTAX:
+            return new FindCommand(fullCommand.substring(FIND_LENGTH));
+        }
         if (isAddCommand(fullCommand)) {
             return new AddCommand(fullCommand);
         }
```
LGTM :100:

wxy1203 commented on src/main/java/duke/Parser.java:
```diff
@@ -104,34 +95,21 @@ public static Command parse(String fullCommand) throws DukeException {
-//            return new UnmarkCommand(Integer.parseInt(fullCommand.substring(UNMARK_LENGTH)));
-//        } else {
-//            throw new DukeException("Parsing error");
-//        }
```
Nice you delete the long tedieous if else if and change it into switch.

wxy1203 reviewed (commented):
Overall, the PR is really good, and your code is understandable. May try to have more explanation on the javadoc to describe the funcitonality of the class or the method.

leehuiyulaura commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,99 @@
+     *
+     * @param filepath location of file stored
+     */
+    private void parseFileToTasks(String filepath) {
```
I believe that this is the longest method in your entire repository and it does not exceed forty lines, thus showing that you have been very successful in avoiding long methods! Nonetheless it could still be improved as the recommended limit for the number of LOC is 30! Still an extremely good effort though! Good Job!

leehuiyulaura commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,87 @@
+     * @param word Word to find in the list of tasks
+     * @return String of the filtered tasks
+     */
+    public String find(String word) {
```
It would be good to avoid such deep nesting of code as evinced by the arrowhead style as seen in this method. Nevertheless I have not seen this frequently throughout your code and you have mostly been able to avoid this problem. Just being nitpicky but would be good to follow!

leehuiyulaura commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,48 @@
+    public Ui() {
+    }
+
+    public void showOutput(String output) {
```
All the method names in your Ui class are very explicit and thus makes it easy for readers to know exactly what the function is meant to do. Good job!

leehuiyulaura commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,115 @@
+     * @param fullCommand input Command from the user.
+     * @return Task object
+     */
+    public static Task commandToTask(String fullCommand) throws DukeException {
```
This method is largely readable and easy to follow and understand!

leehuiyulaura commented on src/main/java/duke/command/ExitCommand.java:
```diff
@@ -0,0 +1,32 @@
+public class ExitCommand extends Command {
+
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) {
```
An overall job well done! All your code is extremely readable and it is easy to follow your logical train of thought.

leehuiyulaura commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,29 @@
+     * @param number 1 based indexing for the task to be deleted.
+     */
+    public DeleteCommand(int number) {
+        this.indexToDelete = number - 1;
```
I like how you add methods even for such simple arithmetic operations as it makes your code more understandable.

leehuiyulaura reviewed (commented):
Very clear, concise, easy to understand and read code. I believe that a lot of effort was put into your code. Job well done! Congratulations.

TanPingZhi committed:
Package tasks

TanPingZhi committed:
Add assertions to command

TanPingZhi committed:
Add switch statements in Parser

Switch statements in Parser helps with better readability and removes
number of helper functions.

TanPingZhi committed:
Merge pull request #1 from TanPingZhi/branch-A-CodeQuality

Branch a code quality

TanPingZhi committed:
Merge branch 'master' into branch-A-Assertions

TanPingZhi committed:
Merge pull request #2 from TanPingZhi/branch-A-Assertions

Branch a assertions

TanPingZhi committed:
Write full commit message for Parser

This is a example full commit message since I forgot.
For commit in a-codequality

Parser was wordy to read as the helper functions are long and
there are many of them.

Add switch statements

Switch statements made Parser more readable and decrease complexity
of the Class

TanPingZhi committed:
Make TaskList sort the tasks

TaskList previously do not sort the tasks according to category or date.

This leads users to pay attention to the wrong tasks.

Create TimeTask class. TimeTask extends Task.

TimeTask allows the differentiation between Todo and tasks with
localDate.

TanPingZhi committed:
Fix bug for gradle build

Currently gradle is unable to build jar file due to missing
javafx but javafx is already included in the dependency.

The bug is due to the main file extending application. Add a wrapper
class file to launch the application.

TanPingZhi committed:
Add error handling

TanPingZhi committed:
Refactor Storage parseFileToTasks() method

TanPingZhi committed:
Add more testing

TanPingZhi committed:
Add docs/Ui.png

TanPingZhi committed:
Update README.md

TanPingZhi committed:
Merge branch 'master' of https://github.com/TanPingZhi/ip

TanPingZhi committed:
Update Ui.png UserGuide and TaskList
